clang/lib/CIR/CodeGen/CIRGenExprCXX.cpp

```diff
@@ -302,7 +302,27 @@ RValue CIRGenFunction::emitCXXMemberOrOperatorMemberCallExpr(
            "Destructor shouldn't have explicit parameters");
     assert(ReturnValue.isNull() && "Destructor shouldn't have return value");
     if (useVirtualCall) {
-      llvm_unreachable("NYI");
+      GlobalDecl globalDecl(dtor, Dtor_Complete);
+
+      const CIRGenFunctionInfo &funcInfo =
+          CGM.getTypes().arrangeCXXStructorDeclaration(globalDecl);
+      cir::FuncType funcTy = CGM.getTypes().GetFunctionType(funcInfo);
+
+      CIRGenCallee callee =
+          CIRGenCallee::forVirtual(CE, globalDecl, This.getAddress(), funcTy);
+
+      if (dtor->isVirtual()) {
```

Comment:
Can you add an extra testcase for when this if is not taken?

Reply:
That check seems redundant. We only reach that point through `if (useVirtualCall) `.  And no useVirtualCall without a virtual destructor. So dtor is virtual by this point.